src/search.c

```diff
@@ -0,0 +1,506 @@
+		file = list->data;
+		list = list->next;
+
+		string_or_die(&untard_file, "%s/%i/Manifest.%s", STATE_DIR, file->last_change,
+						file->filename);
+
+		string_or_die(&tarfile, "%s/%i/Manifest.%s.tar", STATE_DIR, file->last_change,
+						file->filename);
+
+		if (access(untard_file, F_OK) == -1) {
+			/* Does not exist client-side. Must download */
+			string_or_die(&url, "%s/%i/Manifest.%s.tar", preferred_content_url,
+							file->last_change, file->filename);
+
+			ret = swupd_query_url_content_size(url);
+			if (ret != -1) {
+				/* Convert file size from bytes to MB */
+				ret = ret/1000000;
+				size_sum += ret;
+			} else {
+				return ret;
+			}
+		}
+	}
+
+	return size_sum;
+}
+
+
+/* download_manifests()
+ * Description: To search Clear bundles for a particular entry, a complete set of
+ *		manifests must be downloaded. This function does so, asynchronously, using
+ *		the curl_multi interface */
+int download_manifests(struct manifest **MoM)
+{
+	struct list *list;
+	struct file *file;
+	char *tarfile, *untard_file, *url;
+	struct manifest *subMan = NULL;
+	int current_version;
+	int ret = 0;
+	double size;
+
+	current_version = read_version_from_subvol_file(path_prefix);
+	swupd_curl_set_current_version(current_version);
+
+	ret = load_manifests(current_version, current_version, "MoM", NULL, MoM);
+	if (ret != 0) {
+		printf("Cannot load official manifest MoM for version %i\n", current_version);
+		return ret;
+	}
+
+	subscription_versions_from_MoM(*MoM, 0);
+
+	list = (*MoM)->manifests;
+	size = query_total_download_size(list);
+	if (size == -1) {
+		printf("Downloading manifests. Expect a delay, up to 100MB may be downloaded\n");
+	} else if (size > 0) {
+		printf("Downloading Clear Linux manifests\n");
+		printf("   %.2f MB total...\n\n", size);
+	}
+
+	while (list) {
+		file = list->data;
+		list = list->next;
+
+		create_and_append_subscription(file->filename);
+
+		string_or_die(&untard_file, "%s/%i/Manifest.%s", STATE_DIR, file->last_change,
+						file->filename);
+
+		string_or_die(&tarfile, "%s/%i/Manifest.%s.tar", STATE_DIR, file->last_change,
+						file->filename);
+
+		if (access(untard_file, F_OK) == -1) {
+			/* Do download */
+			printf(" '%s' manifest...\n", file->filename);
+
+			ret = load_manifests(current_version, file->last_change, file->filename, NULL, &subMan);
+			if (ret != 0) {
+				printf("Cannot load official manifest MoM for version %i\n", current_version);
+			}
+
+			free_manifest(subMan);
+		}
+
+		if (access(untard_file, F_OK) == -1) {
+			string_or_die(&url, "%s/%i/Manifest.%s.tar", preferred_content_url, current_version,
+							file->filename);
+
+			printf("Error: Failure reading from %s\n", url);
+			free(url);
+		}
+
+		unlink(tarfile);
+		free(untard_file);
+		free(tarfile);
+	}
+
+	return ret;
+}
+
+int search_main(int argc, char **argv)
+{
+	int ret = 0;
+	int lock_fd;
+	struct manifest *MoM = NULL;
+
+	if (!parse_options(argc, argv) ||
+		create_required_dirs()) {
+		free_globals();
```

Comment:
init_globals() is called the first time a few lines down, so you'll want to avoid calling it here.